src/coreclr/jit/compiler.h

```diff
@@ -215,17 +215,26 @@ class LclSsaVarDsc
     // The GT_ASG node that generates the definition, or nullptr for definitions
     // of uninitialized variables.
     GenTreeOp* m_asg;
+    // Number uses of this SSA def (may be an over-estimate). Includes phi args uses.
+    unsigned m_numUses;
+    // True if there may be phi args uses of this def
+    // (false implies all uses are non-phi)
+    bool m_hasPhiUse;
+    // True if there may be uses of the def in a different block
+    bool m_hasGlobalUse;
 
 public:
-    LclSsaVarDsc() : m_block(nullptr), m_asg(nullptr)
+    LclSsaVarDsc() : m_block(nullptr), m_asg(nullptr), m_numUses(0), m_hasPhiUse(false), m_hasGlobalUse(false)
```

Comment:
Nit: the duplication of default values in constructors for these can be avoided with member initializers, i. e.
```
    // The GT_ASG node that generates the definition, or nullptr for definitions
    // of uninitialized variables.
    GenTreeOp* m_asg = nullptr;
```
etc.

Reply:
Always forget about those newfangled C++ features.

Reply:
Unfortunately, you can't do this yet for bitfields, at least not without C++ 20 support.